framework/include/loops/ComputeJacobianForScalingThread.h

```diff
@@ -0,0 +1,29 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ComputeJacobianThread.h"
+
+#include "libmesh/threads.h"
+#include "libmesh/elem_range.h"
+
+class FEProblemBase;
+typedef unsigned int TagID;
+
```

Comment:
We have "TagID" defined in some header already. 

#include "MooseTypes.h"

Reply:
I generally like to avoid header dependencies when they are not needed. It can add significantly to compile times. I understand that this makes it harder to change the `typedef` in the future, but APIs shouldn't be that easy to change, should they? 😄

Reply:
Yeah - I'm not sure if having the same type defined in multiple places is a great idea even if it's with the best intentions. How about splitting up moose types so that we have a file without any includes in it (where that's possible) and then the rest in a different file? I'm open to other ideas here. I also don't think this is a showstopper.

Reply:
I've decided to just include `MooseTypes` here. That header really does a pretty good job of having minimal includes itself. It's just that people are throwing things in `MooseTypes` all the time so you'll have to recompile this a lot, but that's ok